internal/mackerel/service_test.go

```diff
@@ -0,0 +1,136 @@
+package mackerel
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mackerelio/mackerel-client-go"
+)
+
+func Test_ServiceNameValidator(t *testing.T) {
+	t.Parallel()
+
+	cases := map[string]struct {
+		val       types.String
+		wantError bool
+	}{
+		"valid": {
+			val: types.StringValue("service1"),
+		},
+		"too short": {
+			val:       types.StringValue("a"),
+			wantError: true,
+		},
+		"too long": {
+			val:       types.StringValue("toooooooooooooooooooo-looooooooooooooooooooooooooooooooooooooooooong"),
+			wantError: true,
+		},
+		"invalid char": {
+			val:       types.StringValue("v('ω')v"),
+			wantError: true,
+		},
+	}
+
+	ctx := context.Background()
+	for name, tt := range cases {
+		tt := tt
```

Comment:
[may] As long as using Go 1.22+, this reassignment may be not required.

https://tip.golang.org/doc/go1.22#:~:text=Previously%2C%20the%20variables,in%20Go%201.21.

Reply:
fixed: https://github.com/mackerelio-labs/terraform-provider-mackerel/pull/214/commits/225479b53b9ca1e60ddd24f4678aae875440c259